modules/cluster/variables.tf

```
@@ -42,6 +42,12 @@ variable "iam_config" {
   description = "The IAM roles used by the cluster, If you use the included IAM module you can provide it's config output variable."
 }
 
+variable "iam_permissions_boundary" {
```

I think this needs a more specific name:
```suggestion
variable "cluster_autoscaler_iam_permissions_boundary" {
```